app/gui2/shared/languageServer/files.ts

```diff
@@ -0,0 +1,33 @@
+import { type LanguageServer } from 'shared/languageServer'
+import type { FileSystemObject, Path } from 'shared/languageServerTypes'
+
+export async function walkFs(
+  ls: LanguageServer,
+  path: Path,
+  cb: (type: FileSystemObject['type'], path: Path) => void,
+) {
+  for (const file of (await ls.listFiles(path)).paths) {
+    const filePath: Path = {
+      rootId: file.path.rootId,
+      segments: [...file.path.segments, file.name],
+    }
+    cb(file.type, filePath)
+    switch (file.type) {
+      case 'Directory':
+      case 'DirectoryTruncated': {
+        await walkFs(ls, filePath, cb)
```

Comment:
Since this function is callback based, is the `await` actually desirable here? I feel like all it does is introducing potentially unnecessary IO serialization.